lib/mixlib/install.rb

```diff
@@ -83,6 +83,34 @@ def install_command
     end
 
     #
+    # Download a single artifact
+    #
+    # @param [String] download directory. Default: Dir.pwd
+    #
+    # @return [String] file path of downloaded artifact
+    #
+    def download_artifact(directory = Dir.pwd)
+      if options.platform.nil? || options.platform_version.nil? || options.architecture.nil?
+        raise "Must provide platform options to download a specific artifact"
+      end
+
+      artifact = artifact_info
+
+      FileUtils.mkdir_p directory
+      file = File.join(directory, File.basename(artifact.url))
+
+      uri = URI.parse(artifact.url)
+      Net::HTTP.start(uri.host) do |http|
```

Comment:
Won't this have issues with proxies since we are calling Net::HTTP rather than OpenURI?

Reply:
`URI.parse` does the things.